functions/src/requests/index.ts

```diff
@@ -101,35 +104,35 @@ export const triggerEventsWhenRequestIsCreated = functions.firestore
   // eslint-disable-next-line no-unused-vars,@typescript-eslint/no-unused-vars
   .onCreate((snapshot: DocumentSnapshot, context: EventContext) => {
     return validateRequest(snapshot.data() as IRequest)
+      .then(() => {
+        return indexRequest(snapshot);
+      })
       .catch(errors => {
         console.error('Invalid Request Found: ', errors);
         return firestore()
           .collection('requests')
           .doc(context.params.requestId)
           .delete();
-      })
-      .then(() => {
-        return indexRequest(snapshot);
       });
   });
 
 export const triggerEventsWhenRequestIsUpdated = functions.firestore
   .document('requests/{requestId}')
   .onUpdate((change: Change<DocumentSnapshot>, context: EventContext) => {
     return validateRequest(change.after.data() as IRequest)
-      .catch(errors => {
-        console.error('Invalid Request Found: ', errors);
-        return firestore()
-          .collection('requests')
-          .doc(context.params.requestId)
-          .delete();
-      })
       .then(() => {
         return Promise.all([
           queueStatusUpdateTriggers(change),
           queueRatingUpdatedTriggers(change),
           indexRequest(change.after),
         ]);
+      })
+      .catch(errors => {
+        console.error('Invalid Request Found: ', errors);
+        return db
+          .collection('requests')
+          .doc(context.params.requestId)
+          .delete();
```

Comment:
Same issue here, the request would be indexed even if the trigger had deleted it.